### ISSUE TITLE
CTE/WITH clause not working in .moor file

### ISSUE THREAD

**Issue by johrpan** [simolus3/moor#255]
I have a select statement in my .moor file that includes a CTE. When I run the generator I get the following warning and the method is not included in the generated database.

```
[WARNING] moor_generator:moor_generator on lib/database.dart:
Error while generating APIs for worksByComposer
type 'CommonTableExpression' is not a subtype of type 'Table' in type cast
```

I also tested it using the example from the parser test:

```sql
WITH RECURSIVE
cnt(x) AS (
    SELECT 1
    UNION ALL
    SELECT x+1 FROM cnt
    LIMIT 1000000
)
SELECT x FROM cnt;
```

...which gives me the same warning. Are CTEs supposed to work in custom select statements?

**Comment by simolus3** [comment]
Thanks for the report! This is a bug in `moor_generator`, so the parsing test didn't cover it unfortunately. It should be fixed now, I'll release an update later today.

**Comment by johrpan** [comment]
This is great news! Thanks a lot for the instant fix and in general for your outstanding work!

**Comment by simolus3** [comment]
Thanks! I've just published `moor_generator` version 2.1.1 which contains the fix.
